File: examples/qpt.py
from qiskit import ClassicalRegister, QuantumCircuit
from qiskit.circuit import Gate

qc = QuantumCircuit(1, 1)
pre = Gate("pre", 1, [])
post = Gate("post", 1, [])

qc.reset(0)
qc.append(pre, [0])
qc.barrier(0)
qc.h(0)
qc.barrier(0)
qc.append(post, [0])
qc.measure(0, 0)

File: examples/qpt.cpp
sim.set_num_qubits(1);
std::vector<int> creg(1);
// Unsupported operation: reset
// Unsupported operation: pre
// Unsupported operation: barrier
sim.hadamard(0, std::vector<int>{}, std::vector<int>{});
// Unsupported operation: barrier
// Unsupported operation: post
creg[0] = sim.measure(0);
